Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -0,0 +1,136 @@
+        ].compactMap { NSClassFromString($0) }
+#if os(iOS)
+        ignoreClasses = [ UISlider.self, UISwitch.self ]
+#else
+        ignoreClasses = []
+#endif
+    }
+    
+    func redactRegionsFor(view: UIView, options: SentryRedactOptions?) -> [RedactRegion] {
+        var redactingRegions = [RedactRegion]()
+        
+        self.mapRedactRegion(fromView: view,
+                             to: view,
+                             redacting: &redactingRegions,
+                             area: view.frame,
+                             redactText: options?.redactAllText ?? true,
+                             redactImage: options?.redactAllImages ?? true)
+        
+        return redactingRegions
+    }
+    
+    private func shouldIgnore(view: UIView) -> Bool {
+        ignoreClasses.contains { view.isKind(of: $0) }
+    }
+    
+    private func shouldRedact(view: UIView, redactText: Bool, redactImage: Bool) -> Bool {
+        if redactImage, let imageView = view as? UIImageView {
+            return shouldRedact(imageView: imageView)
+        }
+        return redactText && redactClasses.contains { view.isKind(of: $0) }
+    }
+    
+    private func shouldRedact(imageView: UIImageView) -> Bool {
+        // Checking the size is to avoid redact gradient backgroud that
+        // are usually small lines repeating
+        guard let image = imageView.image, image.size.width > 10 && image.size.height > 10  else { return false }
+        return image.imageAsset?.value(forKey: "_containingBundle") == nil
+    }
+    
+    private func mapRedactRegion(fromView view: UIView, to: UIView, redacting: inout [RedactRegion], area: CGRect, redactText: Bool, redactImage: Bool) {
+        let rectInWindow = view.convert(view.bounds, to: to)
+        guard (redactImage || redactText) && area.intersects(rectInWindow) && !view.isHidden && view.alpha != 0 else { return }
+        
+        let ignore = shouldIgnore(view: view)
+        let redact = shouldRedact(view: view, redactText: redactText, redactImage: redactImage)
+        
+        if !ignore && redact {
+            redacting.append(RedactRegion(rect: rectInWindow, color: self.color(for: view)))
+            return
+        } else if hasBackground(view) {
+            if rectInWindow == area {
+                redacting.removeAll()
+            } else {
+                redacting = redacting.flatMap { $0.splitBySubtracting(region: rectInWindow) }
+            }
+        }
+        
+        if !ignore {
+            for subview in view.subviews {
+                mapRedactRegion(fromView: subview, to: to, redacting: &redacting, area: area, redactText: redactText, redactImage: redactImage)
+            }
+        }
+    }
+    
+    private func color(for view: UIView) -> UIColor? {
+        return (view as? UILabel)?.textColor
+    }
+    
+    private func hasBackground(_ view: UIView) -> Bool {
+        return view.backgroundColor != nil && (view.backgroundColor?.cgColor.alpha ?? 0) > 0.9
```

Comment:
`m`: Why only if the alpha is greater than `0.9`? Can you add this info as a comment?